### ISSUE TITLE
Update tcell to the latest 2.x version

### ISSUE THREAD

**Issue by mum4k** [mum4k/termdash#254]
Changelog: https://github.com/gdamore/tcell/blob/master/CHANGESv2.adoc

And while we are at it, we can also consider making `tcell` the default in all code examples.

**Comment by mum4k** [comment]
@dyc3 I got some prep work done (making tcell default, updating documentation), but didn't get to create the PR that updates to 2.x yet.

The next steps should be something like:
1. update the dependency
1. update all import paths
1. verify that none of the [breaking changes](https://github.com/gdamore/tcell/blob/master/CHANGESv2.adoc#breaking-changes-in-tcell-v2) actually break our build and functionality. The only one that looks worrying is the change of mouse button handling. We may need to use or modify one of the demos to verify the new behavior.

If you have the time to prepare the PR with at least the first two steps (and anything I might have missed), that would help a lot. Alternatively I can get to it a few days from now. I will be of course happy to help with the testing step, maybe we can both poke at it - more eyes gives us a better chance of catching anything that broke.

**Comment by mum4k** [comment]
@dyc3 if you choose to work on the PR, please fork from the **devel** branch to avoid conflicts and include the latest changes we made.

**Comment by dyc3** [comment]
I'll take a look at it tomorrow. It's currently 3 am where I am and I need sleep haha

**Comment by mum4k** [comment]
We seem to be in the same timezone, so I can relate...

**Comment by dyc3** [comment]
So far I've only updated the import path for tcell to reference v2, and all the tests pass. Seems like a good sign.

**Comment by mum4k** [comment]
While that is good news, it is likely that it only gives a false sense of security. The unit tests in `termdash` only verify its own logic and only reach up to the [terminalapi](https://github.com/mum4k/termdash/tree/master/terminal/terminalapi). The terminal drivers themselves are abstracted behind a [fake terminal](https://github.com/mum4k/termdash/tree/master/private/faketerm) implementation in the tests.

If I remember correctly, we don't have a single integration test with either `termbox` or `tcell`, so the only way to tell if this works correctly may be by running the demos and exercising the functionality.

**Comment by dyc3** [comment]
Cell options are broken. Nothing has color, and I'm not exactly sure why. Font modifiers still work though. I think the way colors were represented was changed?

**Comment by mum4k** [comment]
Thanks for catching that, I do remember reading something about color palette changes in the v2 changelog. We may need to update our tcell library to do any necessary translation.

Let me know if you get tired of looking at it, I can check out your branch and try to propose some solutions.

**Comment by dyc3** [comment]
I'm kinda stumped.

I think `fixColor` needs to add `tcell.ColorValid` to the color value? When I try that, I get colors back, but they are all wrong.

**Comment by mum4k** [comment]
I will take a look a bit later tonight and update here. I need to spend some time understanding the change they did to v2.

**Comment by mum4k** [comment]
I think I understand the problem now and I am fairly convinced this is a bug in our `tcell` library. I will send you a PR branched off your branch that will fix that and also align our color definitions with the standard `tcell` is using. Since that is the official standard. I just need to figure out how to make this backward compatible for the `termbox` users out there.

**Comment by mum4k** [comment]
The following PR is addressing the color problem: https://github.com/dyc3/termdash/pull/1

The issue was that our tcell library was depending on the **values** of `tcell`'s color constants. We shouldn't have done that, those are internal to their implementation and they did change them when upgrading to v2.

Feel free to merge these commits into your PR, we can push it into the devel branch altogether.

Did you find any other incompatible or breaking changes? Any concerns about the change related to the middle mouse button?

**Comment by dyc3** [comment]
I'll go ahead and merge that in.

I haven't tested the mouse button stuff yet. I'll look at it in a bit, I'm currently fighting fires on another project.

**Comment by mum4k** [comment]
Take your time and feel free to let me know if you would like me to take point on this.

**Comment by dyc3** [comment]
I've tested mouse inputs on Linux, and they work as expected. I don't have an easy way to test on windows currently.

The function keys also work as expected, however modifiers are not passed along at all. Is this intentional?

**Comment by mum4k** [comment]
Thank you for testing it, I will give it a go too to confirm later today.

Can you give me a few more details about the modifiers? Do you mean that some of our keys don't work, e.g. `KeyCtrlA`:
https://github.com/mum4k/termdash/blob/2a7dafa3a835dbe45d9a6eaa2f22e184666c8065/keyboard/keyboard.go#L59

Or do you mean that the new support for modifiers added by `tcell` in v2 isn't reflected in our API?
https://github.com/gdamore/tcell/blob/7d87d8188c8d3a236c778107d4925a9b2465a5f6/key.go#L73-L79

In other words - did any existing functionality change after upgrading `tcell` to v2 or are we just missing new functionality?

Additionally is #260 ready for review or do we still need to add something?

**Comment by dyc3** [comment]
Oh wait, I see. 
The keys like `KeyCtrlA` work, but <kbd>Shift</kbd>+<kbd>F1</kbd> gives `KeyF1`.

We're just missing new functionality. #260 is ready to review

**Comment by mum4k** [comment]
Yes that would be a missing functionality and is tracked in #262.

The PR looks good, please let me know if you are planning to send one more PR to add italic and strikethrough before we push the release.

**Comment by mum4k** [comment]
Resolved by #266.
